src/libbson/src/bson/bson-utf8.c

```diff
@@ -230,6 +231,116 @@ bson_utf8_validate (const char *utf8, /* IN */
    return true;
 }
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * _is_special_char --
+ *
+ *       Uses a bit mask to check if a character requires special formatting
+ *       or not. Called from bson_utf8_escape_for_json.
+ *
+ * Parameters:
+ *       @c: An unsigned char c.
+ *
+ * Returns:
+ *       true if @c requires special formatting. otherwise false.
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+static BSON_INLINE bool
+_is_special_char (unsigned char c)
+{
+   /*
+   C++ equivalent:
+   std::bitset<256> charmap = [...]
+   return charmap[c];
+   */
+   static const bson_unichar_t charmap[8] = {0xffffffff, // control characters
+                                             0x00000004, // double quote "
+                                             0x10000000, // backslash
+                                             0x00000000,
+                                             0xffffffff,
+                                             0xffffffff,
+                                             0xffffffff,
+                                             0xffffffff}; // non-ASCII
+   const int int_index = ((int) c) / ((int) sizeof (bson_unichar_t) * 8);
+   const int bit_index = ((int) c) & ((int) sizeof (bson_unichar_t) * 8 - 1);
+   return ((charmap[int_index] >> bit_index) & ((bson_unichar_t) 1)) != 0u;
+}
+
+/*
+ *--------------------------------------------------------------------------
+ *
+ * _bson_utf8_handle_special_char --
+ *
+ *       Appends a special character in the correct format when converting
+ *       from UTF-8 to JSON. This includes characters that should be escaped
+ *       as well as ASCII control characters.
+ *
+ *       Normal ASCII characters are handled in bson_utf8_escape_for_json,
```

Comment:
```suggestion
 *       Normal ASCII characters and multi-byte UTF-8 sequences are handled in bson_utf8_escape_for_json,
```

Suggest noting this function does not apply to multi-byte UTF-8 sequences. `_is_special_char` returns true for the first byte of a multi-byte UTF-8 sequence, but multi-byte UTF-8 sequences are not expected to be passed to `_bson_utf8_handle_special_char`